changelogs/unreleased/241-handle-max-clients-limit-in-agents.yml

```diff
@@ -0,0 +1,7 @@
+---
+description: Handle max-clients limit in agents better by adding a config parameter to set the limit, raising a warning if the pool gets exhausted and putting the heartbeat in a different pool.
```

Comment:
I would definitely mention that we added a new configuration parameter as you did, as for the 2 other parts I'm not sure how relevant it is for the end-user, @sanderr wdyt ?